lib/Workflow/Condition.pm

```diff
@@ -25,6 +29,140 @@ sub evaluate {
     croak "Class ", ref($self), " must implement 'evaluate()'!\n";
 }
 
+
+sub evaluate_condition {
+    my ( $class, $wf, $condition_name) = @_;
+    $log ||= get_logger();
+    $wf->type;
+
+    my $factory = $wf->_factory();
+    my $orig_condition = $condition_name;
+    my $opposite       = 0;
+    my $condition;
+
+    $log->is_debug
+        && $log->debug("Checking condition $condition_name");
+
+    if ( $condition_name =~ m{ \A ! }xms ) {
+
+        # this condition starts with a '!' and is thus supposed
+        # to return the opposite of an original condition, whose
+        # name is the same except for the '!'
+        $orig_condition =~ s{ \A ! }{}xms;
+        $opposite = 1;
+        $log->is_debug
+            && $log->debug(
+            "Condition starts with a !: '$condition_name'");
+    }
+
+    local $wf->{'_condition_result_cache'} =
+        $wf->{'_condition_result_cache'} || {};
+    if ( $Workflow::Condition::CACHE_RESULTS
+         && exists $wf->{'_condition_result_cache'}->{$orig_condition} ) {
+
+        # The condition has already been evaluated and the result
+        # has been cached
+        $log->is_debug
+            && $log->debug(
+            "Condition has been cached: '$orig_condition', cached result: ",
+            $wf->{'_condition_result_cache'}->{$orig_condition}
+            );
+        if ( !$opposite ) {
+            $log->is_debug
+                && $log->debug("Opposite is false.");
+            if ( !$wf->{'_condition_result_cache'}->{$orig_condition} )
+            {
+                $log->is_debug
+                    && $log->debug("Cached condition result is false.");
+            }
+            return $wf->{'_condition_result_cache'}->{$orig_condition};
+        } else {
+
+            # we have to return an error if the original cached
+            # condition did NOT fail
+            $log->is_debug
+                && $log->debug("Opposite is true.");
+            if ( $wf->{'_condition_result_cache'}->{$orig_condition} ) {
+                $log->is_debug
+                    && $log->debug("Cached condition is true.");
+                return 0;
+            }
+            return 1;
+        }
+    } else {
+
+        # we did not evaluate the condition yet, we have to do
+        # it now
+        $condition = $wf->_factory()
+            ->get_condition( $orig_condition, $wf->type );
+        $log->is_debug
+            && $log->debug( q{Evaluating condition '},
```

Comment:
I am not too crazy about the existing style used in workflow, which mixes single and double quotes, perhaps a strategy using `sprintf` should be applied, but this line is not consistent with existing style IMHO

Reply:
Same as the others: copy/pasted from the original file. I've changed it to be in line with what we use elsewhere.